### ISSUE TITLE
kubelet won't restart after reboot - Unable to register node with API server: connection refused 

### ISSUE THREAD

**Issue by maheshmahadevan** [kubernetes/kubeadm#1082]
## Is this a request for help?
Yes , I am opening a new issue since the https://github.com/kubernetes/kubeadm/issues/1026 issue was closed , but it still seems to affect a lot more people

If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.

If no, delete this section and continue on.

## What keywords did you search in kubeadm issues before filing this one?
Error messages in journalctl

If you have found any duplicates, you should instead reply there and close this page.
The other duplicate is closed.

## Is this a BUG REPORT or FEATURE REQUEST?
BUG REPORT

## Versions

**kubeadm version** (use `kubeadm version`):
kubeadm version: &version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-07T23:14:39Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}

**Environment**:
- **Kubernetes version** (use `kubectl version`): 1.11.2 , cant access Kubectl 
- **Cloud provider or hardware configuration**: Virtual Box installed on Windows Server 2012
- **OS** (e.g. from /etc/os-release): Centos &
- **Kernel** (e.g. `uname -a`): Linux localhost.localdomain 3.10.0-862.el7.x86_64 #1 SMP Fri Apr 20 16:44:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- **Others**:

## What happened?
Issue is exactly same as - https://github.com/kubernetes/kubeadm/issues/1026 , I have followed all the steps mentioned including kubeadm reset and kubeadm init , but issue still persists. The issue occured post reboot.

## What you expected to happen?
Kubernetes to respond

## How to reproduce it (as minimally and precisely as possible)?
Same steps as https://github.com/kubernetes/kubeadm/issues/1026

## Anything else we need to know?
https://github.com/kubernetes/kubeadm/issues/1026

**Comment by rosti** [comment]
Hi @maheshmahadevan and thanks for filing this issue.

Can you check if the bridge traffic is flowing through the firewall?

```
cat /proc/sys/net/bridge/bridge-nf-call-iptables
cat /proc/sys/net/bridge/bridge-nf-call-ip6tables
```

**Comment by maheshmahadevan** [comment]
Here is the output from both the commands you requested

```
$ cat /proc/sys/net/bridge/bridge-nf-call-iptables
1
$ cat /proc/sys/net/bridge/bridge-nf-call-ip6tables
1
```

**Comment by neolit123** [comment]
please post logs from the kubelet to verify that this is the same as:
#1026

```
sudo systemctl status kubelet
journalctl -xeu kubelet
```

#1026 was unresolved, so we cannot help much based on that.

something else to not is that after a reboot it might take a few minutes before a cluster can come back up.
`watch kubectl get nodes`

**Comment by rosti** [comment]
Managed to reproduce the issue on a local CentOS VM following the guide. It appears that after restart SELinux is activated and it messes up the container mounts.

The exact cause of failure is an error in the API server:

```
Error: unable to load server certificate: open /etc/kubernetes/pki/apiserver.crt: no such file or directory
```

The certs, static PODs and mounts in the actual containers were OK, but the API server was unable to access the file.

The issue is resolved by permanently disabling SELinux via `/etc/sysconfig/selinux`.

We need to update the getting started guide for CentOS though.

/kind documentation

**Comment by maheshmahadevan** [comment]
Unfourtunately the above change did not work for me, attaching the selinux config.

```
$ cat /etc/sysconfig/selinux

# This file controls the state of SELinux on the system.
# SELINUX= can take one of these three values:
#     enforcing - SELinux security policy is enforced.
#     permissive - SELinux prints warnings instead of enforcing.
#     disabled - No SELinux policy is loaded.
SELINUX=disabled
# SELINUXTYPE= can take one of three two values:
#     targeted - Targeted processes are protected,
#     minimum - Modification of targeted policy. Only selected processes are protected.
#     mls - Multi Level Security protection.
SELINUXTYPE=targeted
```

I tried kubeadm reset and kubeadm init several times and still get same results

```
$ kubeadm init --pod-network-cidr=10.244.0.0/16
[init] using Kubernetes version: v1.11.2
[preflight] running pre-flight checks
I0830 11:17:38.797172    2292 kernel_validator.go:81] Validating kernel version
I0830 11:17:38.797428    2292 kernel_validator.go:96] Validating kernel config
        [WARNING SystemVerification]: docker version is greater than the most recently validated version. Docker version: 18.06.1-ce. Max validated version: 17.03
[preflight/images] Pulling images required for setting up a Kubernetes cluster
[preflight/images] This might take a minute or two, depending on the speed of your internet connection
[preflight/images] You can also perform this action in beforehand using 'kubeadm config images pull'
[kubelet] Writing kubelet environment file with flags to file "/var/lib/kubelet/kubeadm-flags.env"
[kubelet] Writing kubelet configuration to file "/var/lib/kubelet/config.yaml"
[preflight] Activating the kubelet service
[certificates] Generated ca certificate and key.
[certificates] Generated apiserver certificate and key.
[certificates] apiserver serving cert is signed for DNS names [localhost.localdomain kubernetes kubernetes.default kubernetes.default.svc kubernetes.default.svc.cluster.local] and IPs [10.96.0.1 10.0.2.15]
[certificates] Generated apiserver-kubelet-client certificate and key.
[certificates] Generated sa key and public key.
[certificates] Generated front-proxy-ca certificate and key.
[certificates] Generated front-proxy-client certificate and key.
[certificates] Generated etcd/ca certificate and key.
[certificates] Generated etcd/server certificate and key.
[certificates] etcd/server serving cert is signed for DNS names [localhost.localdomain localhost] and IPs [127.0.0.1 ::1]
[certificates] Generated etcd/peer certificate and key.
[certificates] etcd/peer serving cert is signed for DNS names [localhost.localdomain localhost] and IPs [10.0.2.15 127.0.0.1 ::1]
[certificates] Generated etcd/healthcheck-client certificate and key.
[certificates] Generated apiserver-etcd-client certificate and key.
[certificates] valid certificates and keys now exist in "/etc/kubernetes/pki"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/admin.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/kubelet.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/controller-manager.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/scheduler.conf"
[controlplane] wrote Static Pod manifest for component kube-apiserver to "/etc/kubernetes/manifests/kube-apiserver.yaml"
[controlplane] wrote Static Pod manifest for component kube-controller-manager to "/etc/kubernetes/manifests/kube-controller-manager.yaml"
[controlplane] wrote Static Pod manifest for component kube-scheduler to "/etc/kubernetes/manifests/kube-scheduler.yaml"
[etcd] Wrote Static Pod manifest for a local etcd instance to "/etc/kubernetes/manifests/etcd.yaml"
[init] waiting for the kubelet to boot up the control plane as Static Pods from directory "/etc/kubernetes/manifests"
[init] this might take a minute or longer if the control plane images have to be pulled

                Unfortunately, an error has occurred:
                        timed out waiting for the condition

                This error is likely caused by:
                        - The kubelet is not running
                        - The kubelet is unhealthy due to a misconfiguration of the node in some way (required cgroups disabled)
                        - No internet connection is available so the kubelet cannot pull or find the following control plane images:
                                - k8s.gcr.io/kube-apiserver-amd64:v1.11.2
                                - k8s.gcr.io/kube-controller-manager-amd64:v1.11.2
                                - k8s.gcr.io/kube-scheduler-amd64:v1.11.2
                                - k8s.gcr.io/etcd-amd64:3.2.18
                                - You can check or miligate this in beforehand with "kubeadm config images pull" to make sure the images
                                  are downloaded locally and cached.

                If you are on a systemd-powered system, you can try to troubleshoot the error with the following commands:
                        - 'systemctl status kubelet'
                        - 'journalctl -xeu kubelet'

                Additionally, a control plane component may have crashed or exited when started by the container runtime.
                To troubleshoot, list all containers using your preferred container runtimes CLI, e.g. docker.
                Here is one example how you may list all Kubernetes containers running in docker:
                        - 'docker ps -a | grep kube | grep -v pause'
                        Once you have found the failing container, you can inspect its logs with:
                        - 'docker logs CONTAINERID'
couldn't initialize a Kubernetes cluster
```
Kubelet journalctl entries , the last few lines are just repeated.

```
-- Unit kubelet.service has begun starting up.
Aug 30 11:17:54 localhost.localdomain kubelet[2417]: Flag --cgroup-driver has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
Aug 30 11:17:54 localhost.localdomain kubelet[2417]: Flag --cgroup-driver has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
Aug 30 11:17:54 localhost.localdomain kubelet[2417]: I0830 11:17:54.898578    2417 server.go:408] Version: v1.11.2
Aug 30 11:17:54 localhost.localdomain kubelet[2417]: I0830 11:17:54.899656    2417 plugins.go:97] No cloud provider specified.
Aug 30 11:17:54 localhost.localdomain kubelet[2417]: E0830 11:17:54.952126    2417 certificate_manager.go:299] Failed while requesting a signed certificate from the master: cannot create certificate signing request: Post https://10.0.2.15:6443/apis/certificates.k8s.io/v1beta1/certificatesigningrequests: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.096477    2417 server.go:648] --cgroups-per-qos enabled, but --cgroup-root was not specified.  defaulting to /
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.099311    2417 container_manager_linux.go:243] container manager verified user specified cgroup-root exists: []
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.099341    2417 container_manager_linux.go:248] Creating Container Manager object based on Node Config: {RuntimeCgroupsName: SystemCgroupsName: KubeletCgroupsName: ContainerRuntime:docker CgroupsPerQOS:true CgroupRoot:/ CgroupDriver:cgroupfs KubeletRootDir:/var/lib/kubelet ProtectKernelDefaults:false NodeAllocatableConfig:{KubeReservedCgroupName: SystemReservedCgroupName: EnforceNodeAllocatable:map[pods:{}] KubeReserved:map[] SystemReserved:map[] HardEvictionThresholds:[{Signal:nodefs.available Operator:LessThan Value:{Quantity:<nil> Percentage:0.1} GracePeriod:0s MinReclaim:<nil>} {Signal:nodefs.inodesFree Operator:LessThan Value:{Quantity:<nil> Percentage:0.05} GracePeriod:0s MinReclaim:<nil>} {Signal:imagefs.available Operator:LessThan Value:{Quantity:<nil> Percentage:0.15} GracePeriod:0s MinReclaim:<nil>} {Signal:memory.available Operator:LessThan Value:{Quantity:100Mi Percentage:0} GracePeriod:0s MinReclaim:<nil>}]} QOSReserved:map[] ExperimentalCPUManagerPolicy:none ExperimentalCPUManagerReconcilePeriod:10s ExperimentalPodPidsLimit:-1 EnforceCPULimits:true}
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.099780    2417 container_manager_linux.go:267] Creating device plugin manager: true
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.100331    2417 state_mem.go:36] [cpumanager] initializing new in-memory state store
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.100731    2417 state_file.go:82] [cpumanager] state file: created new state file "/var/lib/kubelet/cpu_manager_state"
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.101400    2417 kubelet.go:274] Adding pod path: /etc/kubernetes/manifests
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.101561    2417 kubelet.go:299] Watching apiserver
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: E0830 11:17:55.138801    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://10.0.2.15:6443/api/v1/pods?fieldSelector=spec.nodeName%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: E0830 11:17:55.139036    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:455: Failed to list *v1.Service: Get https://10.0.2.15:6443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: E0830 11:17:55.139103    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:464: Failed to list *v1.Node: Get https://10.0.2.15:6443/api/v1/nodes?fieldSelector=metadata.name%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.146367    2417 client.go:75] Connecting to docker on unix:///var/run/docker.sock
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.146460    2417 client.go:104] Start docker client with request timeout=2m0s
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: W0830 11:17:55.151549    2417 docker_service.go:545] Hairpin mode set to "promiscuous-bridge" but kubenet is not enabled, falling back to "hairpin-veth"
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.151585    2417 docker_service.go:238] Hairpin mode set to "hairpin-veth"
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: W0830 11:17:55.152079    2417 cni.go:172] Unable to update cni config: No networks found in /etc/cni/net.d
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: W0830 11:17:55.174957    2417 hostport_manager.go:68] The binary conntrack is not installed, this can cause failures in network connection cleanup.
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: W0830 11:17:55.175442    2417 cni.go:172] Unable to update cni config: No networks found in /etc/cni/net.d
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.175535    2417 docker_service.go:253] Docker cri networking managed by cni
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.252743    2417 docker_service.go:258] Docker Info: &{ID:45Y7:AFC4:MNGD:7WHI:WOCB:4IQM:X5JQ:KCYR:K6PR:OUPS:KWI7:XGBR Containers:4 ContainersRunning:1 ContainersPaused:0 ContainersStopped:3 Images:11 Driver:overlay2 DriverStatus:[[Backing Filesystem xfs] [Supports d_type true] [Native Overlay Diff true]] SystemStatus:[] Plugins:{Volume:[local] Network:[bridge host macvlan null overlay] Authorization:[] Log:[awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog]} MemoryLimit:true SwapLimit:true KernelMemory:true CPUCfsPeriod:true CPUCfsQuota:true CPUShares:true CPUSet:true IPv4Forwarding:true BridgeNfIptables:true BridgeNfIP6tables:true Debug:false NFd:31 OomKillDisable:true NGoroutines:50 SystemTime:2018-08-30T11:17:55.182241942+05:30 LoggingDriver:json-file CgroupDriver:cgroupfs NEventsListener:0 KernelVersion:3.10.0-862.el7.x86_64 OperatingSystem:CentOS Linux 7 (Core) OSType:linux Architecture:x86_64 IndexServerAddress:https://index.docker.io/v1/ RegistryConfig:0xc420df0380 NCPU:2 MemTotal:8202018816 GenericResources:[] DockerRootDir:/var/lib/docker HTTPProxy: HTTPSProxy: NoProxy: Name:localhost.localdomain Labels:[] ExperimentalBuild:false ServerVersion:18.06.1-ce ClusterStore: ClusterAdvertise: Runtimes:map[runc:{Path:docker-runc Args:[]}] DefaultRuntime:runc Swarm:{NodeID: NodeAddr: LocalNodeState:inactive ControlAvailable:false Error: RemoteManagers:[] Nodes:0 Managers:0 Cluster:<nil>} LiveRestoreEnabled:false Isolation: InitBinary:docker-init ContainerdCommit:{ID:468a545b9edcd5932818eb9de8e72413e616e86e Expected:468a545b9edcd5932818eb9de8e72413e616e86e} RuncCommit:{ID:69663f0bd4b60df09991c08812a60108003fa340 Expected:69663f0bd4b60df09991c08812a60108003fa340} InitCommit:{ID:fec3683 Expected:fec3683} SecurityOptions:[name=seccomp,profile=default]}
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.252967    2417 docker_service.go:271] Setting cgroupDriver to cgroupfs
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.390341    2417 kuberuntime_manager.go:186] Container runtime docker initialized, version: 18.06.1-ce, apiVersion: 1.38.0
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.392985    2417 csi_plugin.go:111] kubernetes.io/csi: plugin initializing...
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.396724    2417 server.go:129] Starting to listen on 0.0.0.0:10250
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.400407    2417 server.go:302] Adding debug handlers to kubelet server.
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: E0830 11:17:55.407088    2417 kubelet.go:1261] Image garbage collection failed once. Stats initialization may not have completed yet: failed to get imageFs info: unable to find data for container /
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.408845    2417 fs_resource_analyzer.go:66] Starting FS ResourceAnalyzer
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.409004    2417 status_manager.go:152] Starting to sync pod status with apiserver
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.409094    2417 kubelet.go:1758] Starting kubelet main sync loop.
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.409126    2417 kubelet.go:1775] skipping pod synchronization - [container runtime is down PLEG is not healthy: pleg was last seen active 2562047h47m16.854775807s ago; threshold is 3m0s]
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.409848    2417 server.go:986] Started kubelet
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.410583    2417 volume_manager.go:247] Starting Kubelet Volume Manager
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.425202    2417 desired_state_of_world_populator.go:130] Desired state populator starts to run
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: E0830 11:17:55.425347    2417 event.go:212] Unable to write event: 'Post https://10.0.2.15:6443/api/v1/namespaces/default/events: dial tcp 10.0.2.15:6443: connect: connection refused' (may retry after sleeping)
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: W0830 11:17:55.443345    2417 cni.go:172] Unable to update cni config: No networks found in /etc/cni/net.d
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: E0830 11:17:55.449918    2417 kubelet.go:2110] Container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.517681    2417 kubelet_node_status.go:269] Setting node annotation to enable volume controller attach/detach
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.522644    2417 kubelet.go:1775] skipping pod synchronization - [container runtime is down]
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.570723    2417 kubelet_node_status.go:79] Attempting to register node localhost.localdomain
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: E0830 11:17:55.572509    2417 kubelet_node_status.go:103] Unable to register node "localhost.localdomain" with API server: Post https://10.0.2.15:6443/api/v1/nodes: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.723662    2417 kubelet.go:1775] skipping pod synchronization - [container runtime is down]
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.775440    2417 kubelet_node_status.go:269] Setting node annotation to enable volume controller attach/detach
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: I0830 11:17:55.780478    2417 kubelet_node_status.go:79] Attempting to register node localhost.localdomain
Aug 30 11:17:55 localhost.localdomain kubelet[2417]: E0830 11:17:55.780886    2417 kubelet_node_status.go:103] Unable to register node "localhost.localdomain" with API server: Post https://10.0.2.15:6443/api/v1/nodes: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: I0830 11:17:56.125521    2417 kubelet.go:1775] skipping pod synchronization - [container runtime is down]
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: E0830 11:17:56.139911    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://10.0.2.15:6443/api/v1/pods?fieldSelector=spec.nodeName%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: E0830 11:17:56.157565    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:455: Failed to list *v1.Service: Get https://10.0.2.15:6443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: E0830 11:17:56.157762    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:464: Failed to list *v1.Node: Get https://10.0.2.15:6443/api/v1/nodes?fieldSelector=metadata.name%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: I0830 11:17:56.189452    2417 kubelet_node_status.go:269] Setting node annotation to enable volume controller attach/detach
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: I0830 11:17:56.199210    2417 kubelet_node_status.go:79] Attempting to register node localhost.localdomain
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: E0830 11:17:56.199789    2417 kubelet_node_status.go:103] Unable to register node "localhost.localdomain" with API server: Post https://10.0.2.15:6443/api/v1/nodes: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: E0830 11:17:56.284803    2417 event.go:212] Unable to write event: 'Post https://10.0.2.15:6443/api/v1/namespaces/default/events: dial tcp 10.0.2.15:6443: connect: connection refused' (may retry after sleeping)
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: I0830 11:17:56.339104    2417 kubelet_node_status.go:269] Setting node annotation to enable volume controller attach/detach
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: I0830 11:17:56.343333    2417 cpu_manager.go:155] [cpumanager] starting with none policy
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: I0830 11:17:56.343363    2417 cpu_manager.go:156] [cpumanager] reconciling every 10s
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: I0830 11:17:56.343419    2417 policy_none.go:42] [cpumanager] none policy: Start
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: Starting Device Plugin manager
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: W0830 11:17:56.346345    2417 manager.go:496] Failed to retrieve checkpoint for "kubelet_internal_checkpoint": checkpoint is not found
Aug 30 11:17:56 localhost.localdomain kubelet[2417]: E0830 11:17:56.347363    2417 eviction_manager.go:243] eviction manager: failed to get get summary stats: failed to get node info: node "localhost.localdomain" not found
Aug 30 11:17:57 localhost.localdomain kubelet[2417]: I0830 11:17:57.000343    2417 kubelet_node_status.go:269] Setting node annotation to enable volume controller attach/detach
Aug 30 11:17:57 localhost.localdomain kubelet[2417]: I0830 11:17:57.005132    2417 kubelet_node_status.go:79] Attempting to register node localhost.localdomain
Aug 30 11:17:57 localhost.localdomain kubelet[2417]: E0830 11:17:57.005597    2417 kubelet_node_status.go:103] Unable to register node "localhost.localdomain" with API server: Post https://10.0.2.15:6443/api/v1/nodes: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:57 localhost.localdomain kubelet[2417]: E0830 11:17:57.108425    2417 certificate_manager.go:299] Failed while requesting a signed certificate from the master: cannot create certificate signing request: Post https://10.0.2.15:6443/apis/certificates.k8s.io/v1beta1/certificatesigningrequests: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:57 localhost.localdomain kubelet[2417]: E0830 11:17:57.141426    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://10.0.2.15:6443/api/v1/pods?fieldSelector=spec.nodeName%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:57 localhost.localdomain kubelet[2417]: E0830 11:17:57.167182    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:464: Failed to list *v1.Node: Get https://10.0.2.15:6443/api/v1/nodes?fieldSelector=metadata.name%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:57 localhost.localdomain kubelet[2417]: E0830 11:17:57.167993    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:455: Failed to list *v1.Service: Get https://10.0.2.15:6443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:58 localhost.localdomain kubelet[2417]: E0830 11:17:58.144798    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://10.0.2.15:6443/api/v1/pods?fieldSelector=spec.nodeName%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:58 localhost.localdomain kubelet[2417]: E0830 11:17:58.172018    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:464: Failed to list *v1.Node: Get https://10.0.2.15:6443/api/v1/nodes?fieldSelector=metadata.name%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:58 localhost.localdomain kubelet[2417]: E0830 11:17:58.172017    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:455: Failed to list *v1.Service: Get https://10.0.2.15:6443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:58 localhost.localdomain kubelet[2417]: I0830 11:17:58.605802    2417 kubelet_node_status.go:269] Setting node annotation to enable volume controller attach/detach
Aug 30 11:17:58 localhost.localdomain kubelet[2417]: I0830 11:17:58.610293    2417 kubelet_node_status.go:79] Attempting to register node localhost.localdomain
Aug 30 11:17:58 localhost.localdomain kubelet[2417]: E0830 11:17:58.610698    2417 kubelet_node_status.go:103] Unable to register node "localhost.localdomain" with API server: Post https://10.0.2.15:6443/api/v1/nodes: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:59 localhost.localdomain kubelet[2417]: E0830 11:17:59.145671    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://10.0.2.15:6443/api/v1/pods?fieldSelector=spec.nodeName%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:59 localhost.localdomain kubelet[2417]: E0830 11:17:59.172538    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:464: Failed to list *v1.Node: Get https://10.0.2.15:6443/api/v1/nodes?fieldSelector=metadata.name%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:17:59 localhost.localdomain kubelet[2417]: E0830 11:17:59.175294    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:455: Failed to list *v1.Service: Get https://10.0.2.15:6443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:18:00 localhost.localdomain kubelet[2417]: E0830 11:18:00.146612    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://10.0.2.15:6443/api/v1/pods?fieldSelector=spec.nodeName%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:18:00 localhost.localdomain kubelet[2417]: E0830 11:18:00.173317    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:464: Failed to list *v1.Node: Get https://10.0.2.15:6443/api/v1/nodes?fieldSelector=metadata.name%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:18:00 localhost.localdomain kubelet[2417]: E0830 11:18:00.179603    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:455: Failed to list *v1.Service: Get https://10.0.2.15:6443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:18:01 localhost.localdomain kubelet[2417]: E0830 11:18:01.147218    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/config/apiserver.go:47: Failed to list *v1.Pod: Get https://10.0.2.15:6443/api/v1/pods?fieldSelector=spec.nodeName%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:18:01 localhost.localdomain kubelet[2417]: E0830 11:18:01.174170    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:464: Failed to list *v1.Node: Get https://10.0.2.15:6443/api/v1/nodes?fieldSelector=metadata.name%3Dlocalhost.localdomain&limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:18:01 localhost.localdomain kubelet[2417]: E0830 11:18:01.180265    2417 reflector.go:205] k8s.io/kubernetes/pkg/kubelet/kubelet.go:455: Failed to list *v1.Service: Get https://10.0.2.15:6443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:18:01 localhost.localdomain kubelet[2417]: E0830 11:18:01.196739    2417 certificate_manager.go:299] Failed while requesting a signed certificate from the master: cannot create certificate signing request: Post https://10.0.2.15:6443/apis/certificates.k8s.io/v1beta1/certificatesigningrequests: dial tcp 10.0.2.15:6443: connect: connection refused
Aug 30 11:18:01 localhost.localdomain kubelet[2417]: W0830 11:18:01.348563    2417 cni.go:172] Unable to update cni config: No networks found in /etc/cni/net.d
Aug 30 11:18:01 localhost.localdomain kubelet[2417]: E0830 11:18:01.348819    2417 kubelet.go:2110] Container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
```

I do see that there seems to be an issue with container network not being ready . I also observered that there are no configuration entries in /etc/cni/net.d folder, i am not sure if there should be any there. I already tried changing the permissions on this folder 
```
Container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
```

**Comment by rosti** [comment]
@maheshmahadevan It appears, that the API server is not running properly. Can you check if there is a running API server and see its logs. You can also fetch the logs of an instance that died recently.

```
# See if there is running API server
docker ps | grep k8s_kube-apiserver

# If there is no running API server, check if there is one that died recently
docker ps -a | grep k8s_kube-apiserver
```

After doing this, get the container ID (the first column) and use `docker logs <container-id>` to fetch the logs.

This can still be due to the above mentioned issue with SELinux if you have simply edited the config file and did not restart the machine or manually stop SELinux via `setenforce 0`.

**Comment by maheshmahadevan** [comment]
Hi, You are right  that the api server is not running properly , but that was the case since the issue started . All the kube pods go into crash loop and then are stuck in pause mode. Here are the logs from the killed container
```
docker logs cef288343ece
Flag --insecure-port has been deprecated, This flag will be removed in a future version.
I0830 07:18:18.215113       1 server.go:703] external host was not specified, using 10.0.2.15
I0830 07:18:18.215513       1 server.go:145] Version: v1.11.2
I0830 07:18:20.259734       1 plugins.go:158] Loaded 8 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,NodeRestriction,Priority,DefaultTolerationSeconds,DefaultStorageClass,MutatingAdmissionWebhook.
I0830 07:18:20.260305       1 plugins.go:161] Loaded 6 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,Priority,PersistentVolumeClaimResize,ValidatingAdmissionWebhook,ResourceQuota.
I0830 07:18:20.263329       1 plugins.go:158] Loaded 8 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,NodeRestriction,Priority,DefaultTolerationSeconds,DefaultStorageClass,MutatingAdmissionWebhook.
I0830 07:18:20.263353       1 plugins.go:161] Loaded 6 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,Priority,PersistentVolumeClaimResize,ValidatingAdmissionWebhook,ResourceQuota.
F0830 07:18:40.273421       1 storage_decorator.go:57] Unable to create storage backend: config (&{ /registry [https://127.0.0.1:2379] /etc/kubernetes/pki/apiserver-etcd-client.key /etc/kubernetes/pki/apiserver-etcd-client.crt /etc/kubernetes/pki/etcd/ca.crt true false 1000 0xc420468a80 <nil> 5m0s 1m0s}), err (dial tcp 127.0.0.1:2379: connect: connection refused)
```

I had done couple of restarts of machine post setting SELinux and when it did not work also done a ```setencforce 0``` to debug the problem. Anything else you want me to try ?

**Comment by rosti** [comment]
@maheshmahadevan the API server can't connect to etcd for some reason. Can you fetch the logs of the etcd container via the above mentioned procedure? Just replace `k8s_kube-apiserver` with `k8s_etcd`.

**Comment by maheshmahadevan** [comment]
I had checked them before , they don't print anything. Here is the output from all possible present etcd containers on my env.
```
[root@localhost net.d]# docker ps -a | grep etcd | grep -v pause
fd94181b8bd8        b8df3b177be2           "etcd --advertise-cl…"   4 minutes ago       Exited (139) 4 minutes ago                            k8s_etcd_etcd-localhost.localdomain_kube-system_385c195b73ae41a39f5cc1078a3c88ee_25
[root@localhost net.d]# docker logs fd94181b8bd8
[root@localhost net.d]# docker ps -a | grep etcd
ab00b4c12ff5        b8df3b177be2           "etcd --advertise-cl…"   13 seconds ago      Exited (139) 12 seconds ago                            k8s_etcd_etcd-localhost.localdomain_kube-system_385c195b73ae41a39f5cc1078a3c88ee_26
296b68de2463        k8s.gcr.io/pause:3.1   "/pause"                 2 hours ago         Up 2 hours                                             k8s_POD_etcd-localhost.localdomain_kube-system_385c195b73ae41a39f5cc1078a3c88ee_0
[root@localhost net.d]# docker logs 296b68de2463
[root@localhost net.d]# docker logs ab00b4c12ff5
[root@localhost net.d]# docker ps | grep etcd
296b68de2463        k8s.gcr.io/pause:3.1   "/pause"                 2 hours ago         Up 2 hours                                   k8s_POD_etcd-localhost.localdomain_kube-system_385c195b73ae41a39f5cc1078a3c88ee_0
[root@localhost net.d]#
```

**Comment by rosti** [comment]
The etcd container gets killed quite early in its life with SIGSEGV. This is quite odd. You may try to add debug level logging to see if that gives any clue what's happening.

To do that, edit `/etc/kubernetes/manifests/etcd.yaml` and add `--debug=true` option to the etcd command line. Then restart the kubelet service and look at the etcd logs.

**Comment by maheshmahadevan** [comment]
I set etcd command to --debug=true and still dont see any logs , i confirmed it was set to true by checking the kubelet journal entries. The VM has enough memory ~6GB left.

**Comment by neolit123** [comment]
@maheshmahadevan just to make sure.
does you VM has more than one CPU core allocated?

**Comment by maheshmahadevan** [comment]
@neolit123 Yes , VM has 2 cores

**Comment by maheshmahadevan** [comment]
Hi, made some progress possibly with regards to the issue. I tried to spin off a new etcd container and run the command on the shell inside container. All i could get was a Segmentation fault.. Please note the error is same with or without the cmd line args. Does this help ?
```
[root@localhost manifests]# docker run -it k8s.gcr.io/etcd-amd64:3.2.18 /bin/sh
/ # ls
bin   dev   etc   home  proc  root  sys   tmp   usr   var
/ # etcd
Segmentation fault
/ # etcd --advertise-client-urls=https://127.0.0.1:2379 --cert-file=/etc/kubernetes/pki/etcd/server.crt --client-cert-auth=true --data-dir=/var/lib/etcd --initial-advertise-peer-urls=https://127.0.0.1:2380 --initial-cluster=l
ocalhost.localdomain=https://127.0.0.1:2380 --key-file=/etc/kubernetes/pki/etcd/server.key --listen-client-urls=https://127.0.0.1:2379 --listen-peer-urls=https://127.0.0.1:2380 --name=localhost.localdomain --peer-cert-file=/e
tc/kubernetes/pki/etcd/peer.crt --peer-client-cert-auth=true --peer-key-file=/etc/kubernetes/pki/etcd/peer.key --peer-trusted-ca-file=/etc/kubernetes/pki/etcd/ca.crt --snapshot-count=10000 --trusted-ca-file=/etc/kubernetes/pk
i/etcd/ca.crt --debug=true
Segmentation fault
/ #
```

**Comment by rosti** [comment]
@maheshmahadevan this seems like a problem with etcd itself. Could be that your CPU is too old, there is a problem with your hypervisor, a memory issue, etc. You can checkout with the etcd issue tracker and see if someone can help you there.

https://github.com/etcd-io/etcd/issues

**Comment by maheshmahadevan** [comment]
@rosti Thanks , i will try to check with etcd issue tracker , but the strange thing is that I ran the cluster for whole 2 days without any issues , the issue came up only after a reboot of the VM.

**Comment by rosti** [comment]
@maheshmahadevan yes, that looks really strange. Could be on disk data corruption. You can try to `docker rm` the etcd image and then `docker pull` it again.

**Comment by maheshmahadevan** [comment]
I somehow managed to get the cluster up and running by following the instructions in https://github.com/kubernetes/kubeadm/issues/881#issuecomment-396638808
i changed the hostname to localhost . I am also not sure what caused to fix my issue though.

I suspect it has something to do with etcd not able to resolve my localhost correctly

**Comment by ManiacDC** [comment]
Sorry for digging up this old issue. I had this same issue. Somebody has registered localhost.<domain>.com on our domain network. Fixing that fixed the issue.

**Comment by wangcheng34** [comment]
I had this same issue. and I get the log of api server is this 

Flag --insecure-port has been deprecated, This flag will be removed in a future version.
I0714 12:17:27.271429       1 server.go:596] external host was not specified, using 192.168.10.14
I0714 12:17:27.271593       1 server.go:150] Version: v1.17.5
I0714 12:17:27.451156       1 plugins.go:158] Loaded 11 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,NodeRestriction,TaintNodesByCondition,Priority,DefaultTolerationSeconds,DefaultStorageClass,StorageObjectInUseProtection,MutatingAdmissionWebhook,RuntimeClass.
I0714 12:17:27.451168       1 plugins.go:161] Loaded 7 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,Priority,PersistentVolumeClaimResize,ValidatingAdmissionWebhook,RuntimeClass,ResourceQuota.
I0714 12:17:27.451806       1 plugins.go:158] Loaded 11 mutating admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,NodeRestriction,TaintNodesByCondition,Priority,DefaultTolerationSeconds,DefaultStorageClass,StorageObjectInUseProtection,MutatingAdmissionWebhook,RuntimeClass.
I0714 12:17:27.451817       1 plugins.go:161] Loaded 7 validating admission controller(s) successfully in the following order: LimitRanger,ServiceAccount,Priority,PersistentVolumeClaimResize,ValidatingAdmissionWebhook,RuntimeClass,ResourceQuota.
I0714 12:17:27.454186       1 client.go:361] parsed scheme: "endpoint"
I0714 12:17:27.454226       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{https://127.0.0.1:2379 0  <nil>}]
I0714 12:17:28.450566       1 client.go:361] parsed scheme: "endpoint"
I0714 12:17:28.450588       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{https://127.0.0.1:2379 0  <nil>}]
panic: context deadline exceeded

goroutine 1 [running]:
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition.NewREST(0xc0006969a0, 0x4f73a80, 0xc00016f200, 0xc00016f428)
	/workspace/anago-v1.17.5-beta.0.65+a04e7d7c202142/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/etcd.go:56 +0x3c1
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver.completedConfig.New(0xc000bc30c0, 0xc0002433c8, 0x50341e0, 0x756f2f0, 0x10, 0x0, 0x0)
	/workspace/anago-v1.17.5-beta.0.65+a04e7d7c202142/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go:147 +0x152b
k8s.io/kubernetes/cmd/kube-apiserver/app.createAPIExtensionsServer(0xc0002433c0, 0x50341e0, 0x756f2f0, 0x0, 0x4f73720, 0xc000b11d00)
	/workspace/anago-v1.17.5-beta.0.65+a04e7d7c202142/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-apiserver/app/apiextensions.go:99 +0x59
k8s.io/kubernetes/cmd/kube-apiserver/app.CreateServerChain(0xc0001e3080, 0xc0000f0840, 0x449c00a, 0xc, 0xc0009a7c48)
	/workspace/anago-v1.17.5-beta.0.65+a04e7d7c202142/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go:183 +0x292
k8s.io/kubernetes/cmd/kube-apiserver/app.Run(0xc0001e3080, 0xc0000f0840, 0x0, 0x0)
	/workspace/anago-v1.17.5-beta.0.65+a04e7d7c202142/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go:152 +0x101
k8s.io/kubernetes/cmd/kube-apiserver/app.NewAPIServerCommand.func1(0xc000820c80, 0xc00081d380, 0x0, 0x1a, 0x0, 0x0)
	/workspace/anago-v1.17.5-beta.0.65+a04e7d7c202142/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go:119 +0x104
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc000820c80, 0xc0000d6010, 0x1a, 0x1b, 0xc000820c80, 0xc0000d6010)
	/workspace/anago-v1.17.5-beta.0.65+a04e7d7c202142/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:826 +0x460
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000820c80, 0x1691a771b5377f9d, 0x7551340, 0xc000076750)
	/workspace/anago-v1.17.5-beta.0.65+a04e7d7c202142/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/workspace/anago-v1.17.5-beta.0.65+a04e7d7c202142/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:864
main.main()
	_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-apiserver/apiserver.go:43 +0xcd
